### PR TITLE
[CI] Refactor compilations workflow

### PR DESCRIPTION
**Pull request by nobu**
Now some entries need multiple variables for customization, and only one environment variable per entry is not enough.
To solve it, dccfff943c3e has introduced overriding variables by `env` key for each entries.
This commit uses `env` keys for the other environment variables too, instead of appending to `$GITHUB_ENV`.

### CODE REVIEW

File: .github/workflows/compilers.yml [hunk 1]
```diff
@@ -63,154 +62,155 @@ jobs:
...
-#         - { key: crosshost, name: s390x-linux-gnu,       value: s390x-linux-gnu, container: crossbuild-essential-s390x }
-#         - { key: crosshost, name: x86_64-w64-mingw32,    value: x86_64-w64-mingw32, container: mingw-w64 }
+#         - { name: aarch64-linux-gnu,     crosshost: aarch64-linux-gnu, container: crossbuild-essential-arm64 }
+#         - { name: arm-linux-gnueabi,     crosshost: arm-linux-gnueabi }
+#         - { name: arm-linux-gnueabihf,   crosshost: arm-linux-gnueabihf }
+#         - { name: i686-w64-mingw32,      crosshost: i686-w64-mingw32 }
+#         - { name: powerpc-linux-gnu,     crosshost: powerpc-linux-gnu }
+#         - { name: powerpc64le-linux-gnu, crosshost: powerpc64le-linux-gnu, container: crossbuild-essential-ppc64el }
+#         - { name: s390x-linux-gnu,       crosshost: s390x-linux-gnu, container: crossbuild-essential-s390x }
+#         - { name: x86_64-w64-mingw32,    crosshost: x86_64-w64-mingw32, container: mingw-w64 }
```
Comment (shyouhei) [inline .github/workflows/compilers.yml:120]: Maybe we can just delete this part...  It seems `crosshost` is just making things complicated.

**Review by shyouhei — COMMENTED** [review]
:+1: for the idea.

File: .github/workflows/compilers.yml (hunk 1 above)
Comment (nobu) [reply .github/workflows/compilers.yml:120]: Remove cross-compilation checks all?

Comment (shyouhei) [reply .github/workflows/compilers.yml:120]: All from this file.  They are commented out due to https://github.com/ruby/ruby/pull/5693 .  I don't think we can fix the issue soon.

Comment (nobu) [reply .github/workflows/compilers.yml:120]: When I tried recently (https://github.com/nobu/ruby/actions/runs/2534101312), thees seem working?

**Comment by junaruga** [comment]
Very good idea!

**Comment by nobu** [comment]
Apart from the crosshost issue, merged for the time being.

**Comment by nobu** [comment]
Oops, forgot to edit the message..

File: .github/workflows/compilers.yml (hunk 1 above)
Comment (shyouhei) [reply .github/workflows/compilers.yml:120]: Seems not working? https://github.com/nobu/ruby/runs/6980800149?check_suite_focus=true#step:12:1011

**Pull request merged**